crashtracker/src/collector/crash_handler.rs

```diff
@@ -674,9 +665,11 @@ pub fn restore_old_handlers(inside_signal_handler: bool) -> anyhow::Result<()> {
     anyhow::ensure!(!prev.is_null(), "No crashtracking previous signal handlers");
     // Safety: The only nonnull pointer stored here comes from Box::into_raw()
     let prev = unsafe { Box::from_raw(prev) };
```

Comment:
We could flip this around, and only rehydrate into a box when we are ready to drop it